nginx/1.25.3-zlib-ng.patch

```diff
@@ -0,0 +1,332 @@
+                    ctx->flush, ctx->redo);
+ 
+-    rc = inflate(&ctx->zstream, ctx->flush);
++    rc = ZPREFIX(inflate)(&ctx->zstream, ctx->flush);
+ 
+     if (rc != Z_OK && rc != Z_STREAM_END && rc != Z_BUF_ERROR) {
+         ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
+@@ -523,7 +530,7 @@
+ 
+     if (rc == Z_STREAM_END && ctx->zstream.avail_in > 0) {
+ 
+-        rc = inflateReset(&ctx->zstream);
++        rc = ZPREFIX(inflateReset)(&ctx->zstream);
+ 
+         if (rc != Z_OK) {
+             ngx_log_error(NGX_LOG_ALERT, r->connection->log, 0,
+@@ -574,7 +581,7 @@
+     ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                    "gunzip inflate end");
+ 
+-    rc = inflateEnd(&ctx->zstream);
++    rc = ZPREFIX(inflateEnd)(&ctx->zstream);
+ 
+     if (rc != Z_OK) {
+         ngx_log_error(NGX_LOG_ALERT, r->connection->log, 0,
+diff -urN nginx-1.25.3.orig/src/http/modules/ngx_http_gzip_filter_module.c nginx-1.25.3/src/http/modules/ngx_http_gzip_filter_module.c
+--- nginx-1.25.3.orig/src/http/modules/ngx_http_gzip_filter_module.c	2023-10-24 15:46:47.000000000 +0200
++++ nginx-1.25.3/src/http/modules/ngx_http_gzip_filter_module.c	2023-11-25 04:01:03.754887799 +0100
+@@ -9,7 +9,14 @@
+ #include <ngx_core.h>
+ #include <ngx_http.h>
+ 
+-#include <zlib.h>
++#if defined(NGX_ZLIB_NG)
++# include <zlib-ng.h>
++# define ZPREFIX(x) zng_ ## x
++# define z_stream zng_stream
++#elif defined(NGX_ZLIB)
++# include <zlib.h>
++# define ZPREFIX(x) x
++#endif
+ 
+ 
+ typedef struct {
+@@ -56,8 +63,7 @@
+     unsigned             done:1;
+     unsigned             nomem:1;
+     unsigned             buffering:1;
+-    unsigned             zlib_ng:1;
+-    unsigned             state_allocated:1;
++    unsigned             intel:1;
+ 
+     size_t               zin;
+     size_t               zout;
+@@ -214,7 +220,7 @@
+ static ngx_http_output_header_filter_pt  ngx_http_next_header_filter;
+ static ngx_http_output_body_filter_pt    ngx_http_next_body_filter;
+ 
+-static ngx_uint_t  ngx_http_gzip_assume_zlib_ng;
++static ngx_uint_t  ngx_http_gzip_assume_intel;
+ 
+ 
+ static ngx_int_t
+@@ -454,7 +460,7 @@
+     ctx->done = 1;
+ 
+     if (ctx->preallocated) {
+-        deflateEnd(&ctx->zstream);
++        ZPREFIX(deflateEnd)(&ctx->zstream);
+ 
+         ngx_pfree(r->pool, ctx->preallocated);
+     }
+@@ -503,32 +509,32 @@
+      * 8K is for zlib deflate_state, it takes
+      *  *) 5816 bytes on i386 and sparc64 (32-bit mode)
+      *  *) 5920 bytes on amd64 and sparc64
+-     *
+-     * A zlib variant from Intel (https://github.com/jtkukunas/zlib)
+-     * uses additional 16-byte padding in one of window-sized buffers.
+      */
+ 
+-    if (!ngx_http_gzip_assume_zlib_ng) {
+-        ctx->allocated = 8192 + 16 + (1 << (wbits + 2))
+-                         + (1 << (memlevel + 9));
++    if (!ngx_http_gzip_assume_intel) {
++        ctx->allocated = 8192 + (1 << (wbits + 2)) + (1 << (memlevel + 9));
```

Comment:
This part (lines 140-175) seems confused.
Nginx already has support for zlib-ng, but this patch seems to overwrite the zlib-ng variant with intel, and the else part with zlib-ng..
This needs to be cleaned up to only actually change what needs to be changed.